packages/flutter_tools/test/commands/build_macos_test.dart

```diff
@@ -64,14 +65,22 @@ void main() {
   testUsingContext('macOS build invokes build script', () async {
     final BuildCommand command = BuildCommand();
     applyMocksToCommand(command);
-    fs.file('macos/build.sh').createSync(recursive: true);
+    fs.directory('macos').createSync();
     fs.file('pubspec.yaml').createSync();
     fs.file('.packages').createSync();
+    final FlutterProject flutterProject = await FlutterProject.fromDirectory(fs.currentDirectory);
+    final Directory flutterBuildDir = fs.directory(getMacOSBuildDirectory());
+
     when(mockProcessManager.start(<String>[
```

Comment:
TODO to refactor eventually to share with `ios/mac.dart`?

Reply:
Done, in build_macos.dart